src/ppxlib.ml

```diff
@@ -1,5 +1,76 @@
 (** Standard library for ppx rewriters *)
 
+(** [ppxlib] is supposed to be opened globally in your PPX source files.
+
+    Opening it will bring several modules in scope, that are useful to have when
+    writing a rewriter:
+
+    - The "core" [ppxlib] modules, such as modules to help manipulate the AST
+      ({!Ast_builder}, {!Ast_pattern}), and a few functions.
+    - Modules from other libraries, such as {!Ast_helper} or {!Pprintast},
+    - The whole AST types (by [including] the {!Ast} module).
+
+    {1 The core [ppxlib] entries} *)
+
+(** {2 Manipulating the AST} *)
+
+module Ast_builder = Ast_builder
+module Ast_pattern = Ast_pattern
+module Ast_traverse = Ast_traverse
+
+(** {2 Context-free rewriting} *)
+
+module Context_free = Context_free
+module Deriving = Deriving
+module Extension = Extension
+module Expansion_context = Expansion_context
+module Code_path = Code_path
+
+(** {2 Other helpers} *)
+
+module Expansion_helpers = Expansion_helpers
+module Merlin_helpers = Merlin_helpers
+module Spellcheck = Spellcheck
+module Keyword = Keyword
+
+(** {2 Driver-related modules} *)
+
+module Driver = Driver
+module Caller_id = Caller_id
+module File_path = File_path
+module Ast_io = Utils.Ast_io.Read_bin
+
+(** {2 Checks} *)
+
+module Attribute = Attribute
+module Reserved_namespaces = Name.Reserved_namespaces
+
+(** {2 Common helper functions} *)
+
+include Common
+
+(** {1 Modules from other libraries}
+
+    Expose some modules from {!Ppxlib_ast}. *)
+
+module Ast = Ppxlib_ast.Ast
+module Ast_helper = Ppxlib_ast.Ast_helper
+module Asttypes = Ppxlib_ast.Asttypes
+module Parse = Ppxlib_ast.Parse
+module Parsetree = Ppxlib_ast.Parsetree
+module Pprintast = Ppxlib_ast.Pprintast
+module Selected_ast = Ppxlib_ast.Selected_ast
+module Location = Location
+module Longident = Longident
+module Loc = Loc
+
+(** {1 The whole AST types} *)
+
+include Ast
+(** Include all the Ast definitions since we need them in every single ppx
+
+    @closed *)
+
```

Comment:
This whole file is a very nice re-ordering! I haven't checked whether some module has gotten lost on the way. Have you double-checked?

Reply:
Yes, I have now double-checked.
Since the compiler-libs include uses destructive substitution, the reordering did not cause any shadowing problem.